backend/src/run.py

```diff
@@ -62,14 +56,36 @@
                 missing_node_count += 1
                 logger.warning(f"Failed to import {module}: {e}")
 
-                # A bit of hardcoding, but for these it's fine
-                # TODO: Could make it so the __init__.py files define what to do when it fails to import
-                if "torch" in str(e).lower() or "facexlib" in str(e).lower():
-                    missing_categories.add(PyTorchCategory.name)
-                elif "onnx" in str(e).lower():
-                    missing_categories.add(ONNXCategory.name)
-                elif "ncnn" in str(e).lower():
-                    missing_categories.add(NCNNCategory.name)
+                # Turn path into __init__.py path
+                init_module = module.split(".")
+                init_module[-1] = "__init__"
+                init_module = ".".join(init_module)
+                try:
+                    category = getattr(importlib.import_module(init_module), "category")
+                    missing_categories.add(category.name)
+                except ImportError as ie:
+                    logger.warning(ie)
+        # Load categories from __init__.py files
+        elif file.endswith(".py") and file == ("__init__.py"):
+            module = os.path.relpath(
+                os.path.join(root, file), os.path.dirname(__file__)
+            )
+            module = module.replace(os.path.sep, ".")[:-3]
+            # Case when it is a single module with a category
+            try:
+                # TODO: replace the category system with a dynamic factory
+                category = getattr(importlib.import_module(module), "category")
+                categories.add(category)
+            except:
+                # Case when it is a folder with multiple modules that has a category order
+                try:
+                    category_order.extend(
+                        getattr(importlib.import_module(module), "category_order")
+                    )
+                except:
+                    pass
```

Comment:
All of this code when you could have just made another file (e.g. `src/nodes/categories.py`) for `category_order` and import that file.

Reply:
True